### DISCUSSION TITLE
Error when minting the NFT

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#3009] [Q&A]
Hello guys,
i have this code:

'''solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;

import {Script, console} from "forge-std/Script.sol";
import {DevOpsTools} from "foundry-devops/src/DevOpsTools.sol";
import {BasicNft} from "../src/BasicNft.sol";

contract MintBasicNft is Script {
    string public constant PUG_URI =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";

    function run() external {
        address mostRecentlyDeployedBasicNft = DevOpsTools.get_most_recent_deployment("BasicNft", block.chainid);
        mintNftOnContract(mostRecentlyDeployedBasicNft);
    }

    function mintNftOnContract(address contractAddress) public {
        vm.startBroadcast();
        BasicNft(contractAddress).mintNft(PUG_URI);
        vm.stopBroadcast();
    }
}

but when running the "make mint ARGS="--network sepolia " " command on foundry, it gives me this error:

Traces:
  [946684] → new MintBasicNft@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← [Return] 4618 bytes of code

  [1103415] MintBasicNft::run()
    ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   └─ ← [Return] [("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol", 1, true, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111", 2, true, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-1733588388.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-1733588510.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-latest.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/31337", 2, true, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/31337/run-1733641836.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/31337/run-1733645942.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/31337/run-1733646077.json", 3, false, false), ("", "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/31337/run-latest.json", 3, false, false)]
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::readFile("/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-1733588388.json") [staticcall]
    │   └─ ← [Return] <file>
    ├─ [0] VM::parseJsonUint("<stringified JSON>", ".timestamp") [staticcall]
    │   └─ ← [Return] 1733588388 [1.733e9]
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[0]") [staticcall]
    │   └─ ← [Return] true
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[0].contractName") [staticcall]
    │   └─ ← [Return] true
    ├─ [0] VM::parseJsonString("<stringified JSON>", "$.transactions[0].contractName") [staticcall]
    │   └─ ← [Return] "BasicNft"
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.transactions[0].contractAddress") [staticcall]
    │   └─ ← [Return] MintBasicNft: [0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519]
    ├─ [0] VM::toString(1) [staticcall]
    │   └─ ← [Return] "1"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[1]") [staticcall]
    │   └─ ← [Return] false
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::readFile("/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-1733588510.json") [staticcall]
    │   └─ ← [Return] <file>
    ├─ [0] VM::parseJsonUint("<stringified JSON>", ".timestamp") [staticcall]
    │   └─ ← [Return] 1733588510 [1.733e9]
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[0]") [staticcall]
    │   └─ ← [Return] true
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[0].contractName") [staticcall]
    │   └─ ← [Return] true
    ├─ [0] VM::parseJsonString("<stringified JSON>", "$.transactions[0].contractName") [staticcall]
    │   └─ ← [Return] "BasicNft"
    ├─ [0] VM::toString(0) [staticcall]
    │   └─ ← [Return] "0"
    ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.transactions[0].contractAddress") [staticcall]
    │   └─ ← [Return] MintBasicNft: [0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519]
    ├─ [0] VM::toString(1) [staticcall]
    │   └─ ← [Return] "1"
    ├─ [0] VM::keyExistsJson("<JSON file>", "$.transactions[1]") [staticcall]
    │   └─ ← [Return] false
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::readFile("/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-nft-f24/broadcast/DeployBasicNft.s.sol/11155111/run-latest.json") [staticcall]
    │   └─ ← [Return] <file>
    ├─ [0] VM::parseJsonUint("<stringified JSON>", ".timestamp") [staticcall]
    │   └─ ← [Return] 1733588510 [1.733e9]
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::startBroadcast()
    │   └─ ← [Return] 
    ├─ [167] MintBasicNft::mintNft("ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json")
    │   └─ ← [Revert] EvmError: Revert
    └─ ← [Revert] EvmError: Revert

Error: script failed: <empty revert data>
make: *** [Makefile:44: mint] Error 1

why?

**Comment by EngrPips** [comment]
Hello @arbdev95, console.log `mostRecentlyDeployedBasicNft` and lets see what will be printed to the terminal.

**Reply by arbdev95** [reply]
solved the problem, it was a problem of network,not on the code

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.
